misc/python/materialize/mzexplore/extract.py

```diff
@@ -82,31 +84,45 @@ def defs(
             item_name = sql.identifier(item["name"])
             fqname = f"{item_database}.{item_schema}.{item_name}"
 
-            if item["type"] == "index":
+            try:
+                item_type = ItemType(item["type"])
+            except ValueError:
+                warn(f"Unsupported item type `{item['type']}` for {fqname}")
+                continue
+
+            create_file = CreateFile(
+                database=item["database"],
+                schema=item["schema"],
+                name=item["name"],
+                item_type=item_type,
+            )
+
+            if create_file.skip():
+                continue
+
+            if item_type == ItemType.INDEX:
```

Comment:
A stupid nit: we could probably have a `show_create_name()` method on `ItemType` that lets us just write `SHOW CREATE {item_type.show_create_name()} {fqname}` and cut this if down to just one branch that might `continue`.

Reply:
Done.